Update for 10.6

jf990 opened the pull request:
Address #1 update for 10.6 core and updated test.

for https://github.com/highlightjs/highlight.js/issues/3008

@joshgoebel I think I got the `dist` firgure out now.

joshgoebel commented on README.md:
````diff
@@ -38,10 +41,14 @@ Mark the code you want to highlight with the curl class:
 
 ```javascript
-hljs.registerLanguage('curl', window.hljsDefineArcade);
+hljs.registerLanguage('curl', window.hljsDefineCurl);
````
This shouldn't need to be mentioned now since the CDN `dist` build does this automagically.

jf990 replied:
good catch, thought I fixed that :(

joshgoebel commented:
Looks pretty good other than my comment.  Nice to see `node_modules` getting nixed. :)

joshgoebel commented on README.md:
```diff
@@ -11,8 +13,9 @@ Include the `highlight.js` script package in your webpage or node app, load this
+<script type="text/javascript" src="/path/to/highlight.min.js"></script>
+<script type="text/javascript" src="/path/to/highlightjs-curl.min.js"></script>
 <script type="text/javascript">
     hljs.registerLanguage('curl', window.hljsDefineCurl);
```
Don't need this either.

jf990 replied:
ok @joshgoebel I decided to take a different approach. my original intent was to give one-stop instructions how to use this, i have users only using curl highlighting and was trying to keep it simple for them, but it doesn't make sense to chase a moving target as the updates on highlight.js are the authoritative way to do it no sense to try to rehash the existing docs and keep them in sync. I ripped it all out and just provided links to the existing docs.

joshgoebel replied:
Ok.  Just now seeing this.  We can see how that works.  It's just a bit different - but I totally see your point.  _Though I wouldn't say we're a very fast moving target_, but we did rename the "start up" JS, so perhaps that's what you're referring to? ;-)

jf990 replied:
well, a lot of the sample start up code I had was from way back in maybe 9 or earlier and wasn't in sync with the current doc. If it changes again then languages like this would have extra maintenance requirements. i prefer single source of truth, but i also want to make it real easy for users to grab and go. tradeoff. i could copy what we have on cypher and change it to curl and just reuse that if you think that's ok.

joshgoebel replied:
> a lot of the sample start up code I had was from way back in maybe 9 or earlier 

Well, 9 is now "very, very old" (in my mind at least) - and was before we even had official 3rd party grammars. We plan to release 11 in the spring/summer. :)  It's expected these things *could* change every major release (yearly-ish), but honestly I don't foresee them changing again for a while.  The only major doc change from 10 -> 11 OTTOMH is updating the API call from `initOnLoad` whatever to the new `highlightAll`...

Although outside things - like using ES6 modules on the client-side might become more popular and require doc changes - but that's not something we're driving, just life in the tech world. :)

Lets go with the reference to core docs and see how it works out. :)

joshgoebel commented on README.md:
```diff
@@ -8,39 +8,25 @@ See https://curl.haxx.se/docs/manpage.html or in your shell use `curl --help` fo
 ## Usage
 
-Include the `highlight.js` script package in your webpage or node app, load this module and register it with `hljs`. Follow instructions at [highlightjs](https://highlightjs.org/) to learn how to include the library and CSS.
+Include the `highlight.js` script package in your webpage or node app, load this module and register it with `hljs`. Follow instructions at [highlightjs](https://highlightjs.org/) to learn how to include the library and CSS. See [Getting started](https://github.com/highlightjs/highlight.js#getting-started) for different integration and module options.
```
That's an interesting choice I didn't see coming... do you think this is better than including your own example of usage (which you've removed below)?

joshgoebel commented on package.json:
```diff
@@ -3,7 +3,7 @@
   "author": "John Foster",
   "description": "Support for using highlight.js to syntax highlight cURL commands.",
-  "main": "src/language/curl.js",
+  "main": "dist/curl.min.js",
```
This isn't right.  This should point to the original module for `npm`, etc...  `dist` is something that someone would manually choose to load in their browser.

joshgoebel replied:
ie, it was correct as-is

jf990 replied:
thanks, i didn't know that

jf990 commented:
@joshgoebel I addressed your concerns above, but this update adds a bit more:

- adds special highlight class for the -X/--request option
- adds relevance for autodetect
- updates the components to be `extra` compatible for testing in the core repo

jf990 commented:
Closing this PR in favor of the 10.7 update
